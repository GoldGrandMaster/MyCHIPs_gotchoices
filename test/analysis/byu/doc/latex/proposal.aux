\relax 
\providecommand\hyper@newdestlabel[2]{}
\providecommand\HyperFirstAtBeginDocument{\AtBeginDocument}
\HyperFirstAtBeginDocument{\ifx\hyper@anchor\@undefined
\global\let\oldcontentsline\contentsline
\gdef\contentsline#1#2#3#4{\oldcontentsline{#1}{#2}{#3}}
\global\let\oldnewlabel\newlabel
\gdef\newlabel#1#2{\newlabelxx{#1}#2}
\gdef\newlabelxx#1#2#3#4#5#6{\oldnewlabel{#1}{{#2}{#3}}}
\AtEndDocument{\ifx\hyper@anchor\@undefined
\let\contentsline\oldcontentsline
\let\newlabel\oldnewlabel
\fi}
\fi}
\global\let\hyper@last\relax 
\gdef\HyperFirstAtBeginDocument#1{#1}
\providecommand\HyField@AuxAddToFields[1]{}
\providecommand\HyField@AuxAddToCoFields[2]{}
\citation{bateman_myCHIPs}
\citation{bateman_myCHIPs}
\@writefile{toc}{\contentsline {section}{\numberline {I}Introduction}{1}{section.1}\protected@file@percent }
\@writefile{toc}{\contentsline {section}{\numberline {II}Driving Example}{2}{section.2}\protected@file@percent }
\@writefile{lof}{\contentsline {figure}{\numberline {1}{\ignorespaces Example Circuit}}{2}{figure.1}\protected@file@percent }
\newlabel{fig:liftProt}{{1}{2}{Example Circuit}{figure.1}{}}
\citation{dill_trace_theory}
\@writefile{toc}{\contentsline {section}{\numberline {III}Thesis}{3}{section.3}\protected@file@percent }
\@writefile{lof}{\contentsline {figure}{\numberline {2}{\ignorespaces Lift Protocol Sequence Example}}{4}{figure.2}\protected@file@percent }
\newlabel{fig:liftSequence}{{2}{4}{Lift Protocol Sequence Example}{figure.2}{}}
\@writefile{toc}{\contentsline {section}{\numberline {IV}Lift Protocol}{4}{section.4}\protected@file@percent }
\citation{bateman_state_machines}
\@writefile{toc}{\contentsline {section}{\numberline {V}Plan of Work}{5}{section.5}\protected@file@percent }
\@writefile{lof}{\contentsline {figure}{\numberline {3}{\ignorespaces Lift States Diagram}}{6}{figure.3}\protected@file@percent }
\newlabel{fig:liftStates}{{3}{6}{Lift States Diagram}{figure.3}{}}
\citation{bitcoin}
\citation{ben_DC}
\citation{SmartContracts}
\citation{Fischer}
\citation{StateMachine}
\citation{Lamport}
\citation{Delzanno_2014}
\citation{Konnov}
\bibstyle{unsrt}
\bibdata{proposal}
\bibcite{bateman_myCHIPs}{1}
\bibcite{dill_trace_theory}{2}
\bibcite{bateman_state_machines}{3}
\bibcite{bitcoin}{4}
\bibcite{ben_DC}{5}
\bibcite{SmartContracts}{6}
\bibcite{Fischer}{7}
\@writefile{toc}{\contentsline {section}{\numberline {VI}Related Work}{7}{section.6}\protected@file@percent }
\@writefile{toc}{\contentsline {section}{References}{7}{section*.1}\protected@file@percent }
\bibcite{StateMachine}{8}
\bibcite{Lamport}{9}
\bibcite{Delzanno_2014}{10}
\bibcite{Konnov}{11}
