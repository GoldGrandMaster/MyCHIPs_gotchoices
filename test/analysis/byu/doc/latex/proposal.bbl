\begin{thebibliography}{10}

\bibitem{bateman_myCHIPs}
Kyle Bateman.
\newblock {\em MyCHIPs Digital Money at
  \url{http://gotchoices.org/mychips/intro.html}}, 2020.

\bibitem{dill_trace_theory}
David~L Dill.
\newblock {\em Trace theory for automatic hierarchical verification of
  speed-independent circuits}, volume~24.
\newblock MIT press Cambridge, MA, 1989.

\bibitem{bateman_state_machines}
Kyle Bateman.
\newblock {\em MyCHIPs Digital Money at
  \url{https://github.com/gotchoices/MyCHIPs/blob/master/doc/Protocol.md}},
  2020.

\bibitem{bitcoin}
Satoshi Nakamoto.
\newblock Bitcoin: A peer-to-peer electronic cash system.
\newblock {\em Cryptography Mailing list at https://metzdowd.com}, 03 2009.

\bibitem{ben_DC}
Benjamin Ogles, Peter Aldous, and Eric Mercer.
\newblock Proving data race freedom in task parallel programs using a weaker
  partial order.
\newblock In {\em 2019 Formal Methods in Computer Aided Design (FMCAD)}, pages
  55--63, 2019.

\bibitem{SmartContracts}
Karthikeyan Bhargavan, Antoine Delignat-Lavaud, C\'{e}dric Fournet, Anitha
  Gollamudi, Georges Gonthier, Nadim Kobeissi, Natalia Kulatova, Aseem Rastogi,
  Thomas Sibut-Pinote, Nikhil Swamy, and Santiago Zanella-B\'{e}guelin.
\newblock Formal verification of smart contracts: Short paper.
\newblock In {\em Proceedings of the 2016 ACM Workshop on Programming Languages
  and Analysis for Security}, PLAS '16, page 91–96, New York, NY, USA, 2016.
  Association for Computing Machinery.

\bibitem{Fischer}
Michael~J. Fischer, Nancy~A. Lynch, and Michael~S. Paterson.
\newblock Impossibility of distributed consensus with one faulty process.
\newblock In {\em Proceedings of the 2nd ACM SIGACT-SIGMOD Symposium on
  Principles of Database Systems}, PODS '83, page 1–7, New York, NY, USA,
  1983. Association for Computing Machinery.

\bibitem{StateMachine}
Fred~B. Schneider.
\newblock Implementing fault-tolerant services using the state machine
  approach: A tutorial.
\newblock {\em ACM Comput. Surv.}, 22(4):299–319, December 1990.

\bibitem{Lamport}
Leslie Lamport.
\newblock Byzantizing paxos by refinement.
\newblock In David Peleg, editor, {\em Distributed Computing}, pages 211--224,
  Berlin, Heidelberg, 2011. Springer Berlin Heidelberg.

\bibitem{Delzanno_2014}
Giorgio Delzanno, Michele Tatarek, and Riccardo Traverso.
\newblock Model checking paxos in spin.
\newblock {\em Electronic Proceedings in Theoretical Computer Science},
  161:131–146, Aug 2014.

\bibitem{Konnov}
Igor Konnov, Helmut Veith, and Josef Widder.
\newblock Who is afraid of model checking distributed algorithms?
\newblock 07 2012.

\end{thebibliography}
