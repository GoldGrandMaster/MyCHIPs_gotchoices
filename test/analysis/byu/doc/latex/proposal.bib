
@inproceedings{Fischer,
author = {Fischer, Michael J. and Lynch, Nancy A. and Paterson, Michael S.},
title = {Impossibility of Distributed Consensus with One Faulty Process},
year = {1983},
isbn = {0897910974},
publisher = {Association for Computing Machinery},
address = {New York, NY, USA},
url = {https://doi.org/10.1145/588058.588060},
doi = {10.1145/588058.588060},
abstract = {The consensus problem involves an asynchronous system of processes, some of which may be unreliable. The problem is for the reliable processes to agree on a binary value. We show that every protocol for this problem has the possibility of nontermination, even with only one faulty process. By way of contrast, solutions are known for the synchronous case, the "Byzantine Generals" problem.},
booktitle = {Proceedings of the 2nd ACM SIGACT-SIGMOD Symposium on Principles of Database Systems},
pages = {1–7},
numpages = {7},
location = {Atlanta, Georgia},
series = {PODS '83}
}
@inproceedings{SmartContracts,
author = {Bhargavan, Karthikeyan and Delignat-Lavaud, Antoine and Fournet, C\'{e}dric and Gollamudi, Anitha and Gonthier, Georges and Kobeissi, Nadim and Kulatova, Natalia and Rastogi, Aseem and Sibut-Pinote, Thomas and Swamy, Nikhil and Zanella-B\'{e}guelin, Santiago},
title = {Formal Verification of Smart Contracts: Short Paper},
year = {2016},
isbn = {9781450345743},
publisher = {Association for Computing Machinery},
address = {New York, NY, USA},
url = {https://doi.org/10.1145/2993600.2993611},
doi = {10.1145/2993600.2993611},
abstract = {Ethereum is a framework for cryptocurrencies which uses blockchain technology to provide an open global computing platform, called the Ethereum Virtual Machine (EVM). EVM executes bytecode on a simple stack machine. Programmers do not usually write EVM code; instead, they can program in a JavaScript-like language, called Solidity, that compiles to bytecode. Since the main purpose of EVM is to execute smart contracts that manage and transfer digital assets (called Ether), security is of paramount importance. However, writing secure smart contracts can be extremely difficult: due to the openness of Ethereum, both programs and pseudonymous users can call into the public methods of other programs, leading to potentially dangerous compositions of trusted and untrusted code. This risk was recently illustrated by an attack on TheDAO contract that exploited subtle details of the EVM semantics to transfer roughly $50M worth of Ether into the control of an attacker.In this paper, we outline a framework to analyze and verify both the runtime safety and the functional correctness of Ethereum contracts by translation to F*, a functional programming language aimed at program verification.},
booktitle = {Proceedings of the 2016 ACM Workshop on Programming Languages and Analysis for Security},
pages = {91–96},
numpages = {6},
keywords = {smart contracts, formal verification, ethereum, EVM, solidity},
location = {Vienna, Austria},
series = {PLAS '16}
}

@article{StateMachine,
author = {Schneider, Fred B.},
title = {Implementing Fault-Tolerant Services Using the State Machine Approach: A Tutorial},
year = {1990},
issue_date = {Dec. 1990},
publisher = {Association for Computing Machinery},
address = {New York, NY, USA},
volume = {22},
number = {4},
issn = {0360-0300},
url = {https://doi.org/10.1145/98163.98167},
doi = {10.1145/98163.98167},
abstract = {The state machine approach is a general method for implementing fault-tolerant services in distributed systems. This paper reviews the approach and describes protocols for two different failure models—Byzantine and fail stop. Systems reconfiguration techniques for removing faulty components and integrating repaired components are also discussed.},
journal = {ACM Comput. Surv.},
month = dec,
pages = {299–319},
numpages = {21}
}

@InProceedings{Lamport,
author="Lamport, Leslie",
editor="Peleg, David",
title="Byzantizing Paxos by Refinement",
booktitle="Distributed Computing",
year="2011",
publisher="Springer Berlin Heidelberg",
address="Berlin, Heidelberg",
pages="211--224",
abstract="We derive a {\$}3f{\backslash}!+{\backslash}!1{\$}process Byzantine Paxos consensus algorithm by Byzantizing a variant of the ordinary Paxos algorithm---that is, by having {\$}2f{\backslash}!+{\backslash}!1{\$}nonfaulty processes emulate the ordinary Paxos algorithm despite the presence of f malicious processes. We have written a formal, machine-checked proof that the Byzantized algorithm implements the ordinary Paxos consensus algorithm under a suitable refinement mapping.",
isbn="978-3-642-24100-0"
}

@article{Delzanno_2014,
   title={Model Checking Paxos in Spin},
   volume={161},
   ISSN={2075-2180},
   url={http://dx.doi.org/10.4204/EPTCS.161.13},
   DOI={10.4204/eptcs.161.13},
   journal={Electronic Proceedings in Theoretical Computer Science},
   publisher={Open Publishing Association},
   author={Delzanno, Giorgio and Tatarek, Michele and Traverso, Riccardo},
   year={2014},
   month={Aug},
   pages={131–146}
}

@inproceedings{Konnov,
author = {Konnov, Igor and Veith, Helmut and Widder, Josef},
year = {2012},
month = {07},
pages = {},
title = {Who is afraid of Model Checking Distributed Algorithms?}
}

@article{bitcoin,
author = {Nakamoto, Satoshi},
year = {2009},
month = {03},
pages = {},
title = {Bitcoin: A Peer-to-Peer Electronic Cash System},
journal = {Cryptography Mailing list at https://metzdowd.com}
}

@article{bateman_myCHIPs, url={http://gotchoices.org/mychips/intro.html}, journal={MyCHIPs Digital Money at \url{http://gotchoices.org/mychips/intro.html} },
 author={Bateman, Kyle},
 year={2020},
 } 
 
@article{bateman_state_machines, url={https://github.com/gotchoices/MyCHIPs/blob/master/doc/Protocol.md}, journal={MyCHIPs Digital Money at \url{https://github.com/gotchoices/MyCHIPs/blob/master/doc/Protocol.md} },
 author={Bateman, Kyle},
 year={2020},
 } 
 
@book{dill_trace_theory,
  title={Trace theory for automatic hierarchical verification of speed-independent circuits},
  author={Dill, David L},
  volume={24},
  year={1989},
  publisher={MIT press Cambridge, MA}
}

@INPROCEEDINGS{ben_DC,  author={Ogles, Benjamin and Aldous, Peter and Mercer, Eric},  booktitle={2019 Formal Methods in Computer Aided Design (FMCAD)},   title={Proving Data Race Freedom in Task Parallel Programs Using a Weaker Partial Order},   year={2019},  volume={},  number={},  pages={55-63},  doi={10.23919/FMCAD.2019.8894270}}
