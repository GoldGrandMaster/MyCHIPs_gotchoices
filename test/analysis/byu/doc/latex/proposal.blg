This is BibTeX, Version 0.99d (TeX Live 2019)
Capacity: max_strings=100000, hash_size=100000, hash_prime=85009
The top-level auxiliary file: proposal.aux
The style file: unsrt.bst
Database file #1: proposal.bib
Warning--empty title in bateman_myCHIPs
Warning--empty title in bateman_state_machines
Warning--empty booktitle in Konnov
You've used 11 entries,
            1791 wiz_defined-function locations,
            523 strings with 5596 characters,
and the built_in function-call counts, 2730 in all, are:
= -- 241
> -- 107
< -- 5
+ -- 41
- -- 30
* -- 160
:= -- 413
add.period$ -- 34
call.type$ -- 11
change.case$ -- 8
chr.to.int$ -- 0
cite$ -- 14
duplicate$ -- 153
empty$ -- 268
format.name$ -- 30
if$ -- 640
int.to.chr$ -- 0
int.to.str$ -- 11
missing$ -- 12
newline$ -- 56
num.names$ -- 13
pop$ -- 46
preamble$ -- 1
purify$ -- 0
quote$ -- 0
skip$ -- 87
stack$ -- 0
substring$ -- 130
swap$ -- 55
text.length$ -- 5
text.prefix$ -- 0
top$ -- 0
type$ -- 0
warning$ -- 3
while$ -- 21
width$ -- 13
write$ -- 122
(There were 3 warnings)
