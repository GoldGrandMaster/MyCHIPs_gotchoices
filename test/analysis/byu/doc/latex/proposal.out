\BOOKMARK [1][-]{section.1}{Introduction}{}% 1
\BOOKMARK [1][-]{section.2}{Driving Example}{}% 2
\BOOKMARK [1][-]{section.3}{Thesis}{}% 3
\BOOKMARK [1][-]{section.4}{Lift Protocol}{}% 4
\BOOKMARK [1][-]{section.5}{Plan of Work}{}% 5
\BOOKMARK [1][-]{section.6}{Related Work}{}% 6
\BOOKMARK [1][-]{section*.1}{References}{}% 7
